src/index.ts

```diff
@@ -103,14 +121,34 @@ export class WebRTCPlayer extends EventEmitter {
     }
   }
 
-  private async onConnectionStats() {
+  private async onConnectionStats() { 
+    
     if (this.peer && this.statsTypeFilter) {
+      let bytesReceivedBlock: number = 0;
       let stats = await this.peer.getStats(null);
+
       stats.forEach((report) => {
         if (report.type.match(this.statsTypeFilter)) {
           this.emit(`stats:${report.type}`, report);
         }
+
+        if (this.detectMediaTimeout == true && report.type.match('inbound-rtp') ) { 
+          this.emit(`stats:${report.type}`, report);
+          bytesReceivedBlock += report.bytesReceived;
+        }
       });
+
+      if (bytesReceivedBlock <= this.bytesReceived) {
+        this.timeoutThresholdCounter += this.msStatsInterval;
```

Comment:
There is no reset for this counter, so if media is lossed from time to time but generally is incoming this will eventually trigger even if media is being sent.